packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js

```diff
@@ -68,9 +70,15 @@ export default function useNestedSettingsUpdate(
 	// render.
 	const _allowedBlocks = useMemo( () => allowedBlocks, allowedBlocks );
 
+	const _inserterPriority = useMemo(
+		() => inserterPriority,
+		inserterPriority
+	);
```

Comment:
There's no reason for this to be memoized.

And unless `useMemo` handles dependencies differently from `useLayoutEffect`, the previous memo does nothing either. If `allowedBlocks` is getting a new object reference each time, the `_allowedBlocks` will also update to the new reference.

Reply:
This is confusing but I think it's correct.

The dependencies list of `useMemo` is an array of dependencies. In this case the dependencies are the string contained within `inserterPriority`.

Without the memo, any non-stable `inserterPriority` array would trigger the layout effect because the reference would change.

With the memo, even if the array reference is unstable the memo will fix it to make it stable. This will avoid the layout effect being called unnecessarily.

The memo "works" because the 2nd argument of the `useMemo` call is the array of strings. As these strings will stay the same even if the `inserterPriority` array references changes the `useMemo` never needs to recompute.

Originally in https://github.com/WordPress/gutenberg/pull/30311/

Reply:
Great explanation! 👏🏻

Reply:
The comment should explain it like that—the existing one isn't very clear when scanning over the code.

Reply:
@ajlende Yeh you have a point there. I can raise a PR to augment the existing comments here if you don't fancy taking it on.